pkgs/applications/virtualization/nvidia-docker/default.nix

```diff
@@ -1,47 +1,63 @@
-{ stdenv, lib, fetchFromGitHub, fetchpatch, callPackage, makeWrapper
-, buildGoPackage, runc, glibc }:
+{ stdenv
+, lib
+, fetchFromGitHub
+, fetchpatch
+, callPackage
+, makeWrapper
+, buildGoModule
+, git
+, glibc
+}:
 
 with lib; let
-
   libnvidia-container = callPackage ./libnvc.nix { };
 
-  nvidia-container-runtime = fetchFromGitHub {
-    owner = "NVIDIA";
-    repo = "nvidia-container-runtime";
-    rev = "runtime-v2.0.0";
-    sha256 = "0jcj5xxbg7x7gyhbb67h3ds6vly62gx7j02zm6lg102h34jajj7a";
-  };
-
-  nvidia-container-runtime-hook = buildGoPackage {
-    pname = "nvidia-container-runtime-hook";
-    version = "1.4.0";
-
-    goPackagePath = "nvidia-container-runtime-hook";
-
-    src = "${nvidia-container-runtime}/hook/nvidia-container-runtime-hook";
+  nvidia-container-runtime = buildGoModule rec {
+    pname = "nvidia-container-toolkit";
+    version = "3.4.0";
+    src = "${fetchFromGitHub {
+      owner = "NVIDIA";
+      repo = "nvidia-container-runtime";
+      rev = "v${version}";
+      sha256 = "095mks0r4079vawi50pk4zb5jk0g6s9idg2s1w55a0d27jkknldr";
+    }}/src";
+    preBuild = "go mod init main";
+    nativeBuildInputs = [ git ];
```

Comment:
```suggestion
```

buildGoModule should already contain git.

Reply:
I changed this to use `buildGoPackage`.